FrontEnd/src/components/Taskoverview.js

```diff
@@ -75,27 +84,52 @@ function Taskoverview(props) {
     }, [])
 
     const taskAccordionsList = taskList.map((task) =>
-        <Accordion key={task.taskId} className={classes.accordion}>
+      {
+      let appropriateAccordionStyle;
+      const currentDate = new Date();
+      let  deadline = new Date(task.deadline);
+
+      //berechnet den Unterschied in Tagen zwischen currentDate und deadline
+      const diffInDays = Math.floor(Date.UTC(deadline.getFullYear(), deadline.getMonth(), deadline.getDate()) -
+        Date.UTC(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate()))  / (1000 * 60 * 60 * 24);
```

Comment:
Bin kein Fan von Magic Numbers. Und kann man die Date Objekte (deadline & currentdate) nicht auch einfach voneinander abziehen?

Reply:
Leider nicht. Hätte ich auch vorgezogen. Aber die Deadline wird im Backend als JSON gespeichert, und das Ist ein String Format, und Strings kann man schlecht subtrahieren.

Das ist zwar Etwas Magie auf den ersten Blick, auf den Zeiten Blick werden hier einfach nur die current time und deadline in UTC Format geparst und diese kann man voneinander abziehen.

Und das Math.floor rundet es nur runter, und die klammer dahinter berechnet dann die Differenz in Tagen.

Reply:
In Zeile 89 & 90 legst du doch zwei Date Objekte an. Die Kannst du safe voneinander subtrahieren. 
Was du danach machst, ist, dass du dein Date Objekt auseinandernimmst um es als Argumente für die Erzeugung eines neuen Date Objekts zu verwenden. 

Magic Numbers ist auf (1000 * 60 * 60 * 24) bezogen

Reply:
const diffInDays = Math.floor( (deadline - currentDate) / (1000 * 60 * 60 *24) )

Reply:
Ok ich hab mit der Funktion nochmal rumgespielt und deinen Ansatz ausprobiert. Was das Problem bei dem const diffInDays = Math.floor( (deadline - currentDate) / (1000 * 60 * 60 *24) ) Ansatz ist, dass hier "Tage" = 24h sind, nicht nur Datums. D.h. Wenn die Aufgabe weniger als 24h übrig hat, dann wird sie rot. Problem hier ist: Du schaust dir die Aufgabe morgens an, sie ist z.B.Geld (Weil noch 26h übrig oder so..) und Abend wäre sie dann rot. 

Während mit meinem etwas...komplizierterem Ansatz, wo ich die Dates erst auseinandernehme um dann die Dates wieder zussammenzubauen, da nehm ich nur Jahr, Monat und Tag. Heißt du kannst dir sicher sein, dass diffInDays, also auch die Farbeanzeige innerhalb eines Datumstages gleich bleibt und sich dann um Mitternacht erst ändert.

Ich denke beides sind Lösungen die zum Ziel führen. Ich würde argumentieren, dass es für einen Schüler nicht so wichtig ist zu wissen ob er noch mehr als 24 oder 36h Zeit hat, sondern das nach Datum wissen muss.  Also Rot heißt ich hab noch heute Zeit an der HA zu arbeiten, da sie morgen fällig ist und ich sie in der Schule abgeben muss. Wenn er statdessen morgens eine gelbe Aufgabe sieht, kann es sein, dass die Aufgabe Am nächsten Tag Mittags abgegeben werden muss (mehr als 24h) , aber durch gelb denkt er, er hat noch einen Tag mehr.

Ich hoffe ich hab rüberbringen können wie ich drüber denke. Wenn du dem nicht zustimmst würde ich das mal auf die nächste Standup tagesordnung packen.

PS.: Falls du bis hierhin durchgehalten hast, sorry für den Roman :D

Reply:
Weitere Anmerkung: das / (1000 * 60 * 60 * 24) ist, da Datumsanzeigen in Millisekunden angegeben werden. /1000 --> sekunden. / 60 --> Minuten, / 60 --> Stunden, /24 --> Tage.
Und wir wollen ja die Differenz in Tagen.

Hätte man auch ausrechnen können die Klammer, aber damit wirds vielleicht etwas weniger eine "Magicnumber"

Reply:
currentDateWithoutTime = new Date(currentDate.toDateString)
deadlineWithoutTime = new Date(deadline.toDateString)

const MILLISECS_ON_A_DAY = 1000 * 60 * 60 *24 
const diffInDays = ( deadlineWithoutTime - currentDateWithoutTime ) / MILLISECS_ON_A_DAY


Das Math.floor müsste sich ja erübrigen, wenn sowieso beides auf 0:00 Uhr steht